### ISSUE TITLE
Unify configuration specification

### ISSUE THREAD

**Issue by travitch** [GaloisInc/crucible#457]
Currently, command line options and config file options are specified separately but with substantial overlap.  It would be really useful to have a single specification produce both sets of options uniformly. There are a few improvements that this would enable/benefits it would provide:

- Less chance of inconsistency between the two specs
- Automatic printing of default values (currently, the description strings have them hard coded, which makes it difficult to change defaults on a per-frontend basis)
- We could extract a user manual section (in e.g., .tex format) describing each option in detail, letting us keep user manuals up-to-date with the tool

@robdockins suggested that we could also use this improved framework to expose simulation-time variables to frontends (e.g., https://github.com/GaloisInc/what4/blob/6189c160b9477fb2bfeb07566d99881eb1c9a202/what4/src/What4/Solver/Yices.hs#L1013).

**Comment by robdockins** [comment]
Yeah there's a pretty sophisticated system for describing and updating configuration options internal to what4.  It's mostly used for configuration options for individual solvers, but is also used for, e.g., the setting describing if path-satisfiability should be on (which means it can be twiddled at runtime)

https://github.com/GaloisInc/what4/blob/master/what4/src/What4/Config.hs

**Comment by langston-barrett** [comment]
Closing in favor of #461
